src/control/security/signature.go

```diff
@@ -56,17 +55,21 @@ func DefaultTokenSigner() *TokenSigner {
 	}
 }
 
-//Sign takes an unhashed set of bytes and hashes and signs the result with the
-//key passed in. If no key is specified it will return the unsigned digest.
-func (s *TokenSigner) Sign(key crypto.PrivateKey, data []byte) ([]byte, error) {
+//Hash returns the SHA-512 hash of the byte array passed in.
+func (s *TokenSigner) Hash(data []byte) ([]byte, error) {
 	hash := sha512.New()
 	if _, err := hash.Write(data); err != nil {
 		return nil, errors.New("hash failed to write")
 	}
-	digest := hash.Sum(nil)
+	return hash.Sum(nil), nil
+}
 
-	if key == nil {
-		return digest, nil
+//Sign takes an unhashed set of bytes and hashes and signs the result with the
+//key passed in. If no key is specified it will return the unsigned digest.
```

Comment:
Comment about no key passed in is no longer accurate

Reply:
Good catch. Thank you. Fixed in the latest version.